lib/taglet/tag_as.ex

```diff
@@ -8,36 +8,57 @@ defmodule Taglet.TagAs do
       Module.register_attribute __MODULE__, :contexts, accumulate: true
       @contexts unquote(context)
 
-      def unquote(:"add_#{singularized_context}")(struct, tag) do
-        Taglet.add(struct, tag, unquote(context))
+      def unquote(:"add_#{singularized_context}")(struct, tag) when is_bitstring(tag) do
```

Comment:
Since we are expecting a string we should use the `is_binary` guard.
A binary has a number of bytes evenly divisible by 8 (does not contain partial bytes). A bitstring does not have this restriction and can contain valid bit strings that are not valid UTF strings. 

https://stackoverflow.com/a/10821192